app/domain/authentication/authn_k8s/k8s_object_lookup.rb

```diff
@@ -134,12 +134,29 @@ def invoke_k8s_method method_name, *arguments
         k8s_client_for_method(method_name).send(method_name, *arguments)
       end
 
+      def k8s_client_info client
+        group    = client.instance_variable_get(:@api_group)
+        version  = client.instance_variable_get(:@api_version)
+        entities = client.instance_variable_get(:@entities).to_yaml
+        "api_group: #{group}, api_version: #{version}, entities: #{entities}"
+      end
```

Comment:
I didn't realize you wanted to use these methods in production code.  They really aren't meant to be used like this.  If the client is making them private, they really should be considered private.

I can't do the research on figured out if there's a more proper way to do this, but just be aware that what we're doing here is very brittle and liable to break with the next client release.  If you've considered all that and still decided this is the only way, let me know.

Reply:
if this part is questionable or merits further discussion, I don't think it's needed to get the client support for new API versions in (right, @diverdane?) and we can separate this out into another PR to discuss there

Reply:
Yes, this isn't necessary for the fix. I'm going to bail on adding the debug level messages for now. I took a look at the kubeclient documentation, and I don't see any good way to expose the "entities" that are supported by each client. (There's an "all_entities" call documented, but it gives me an empty result. :( ).